pkg/libinit/proc_linux.go

```diff
@@ -78,11 +163,67 @@ func RunCommands(debug func(string, ...interface{}), commands ...*exec.Cmd) int
 
 		cmdCount++
 		debug("Trying to run %v", cmd)
+
+		// Set up PTM
+		m, s, err := NewPTMS()
+		if err != nil {
+			log.Printf("Error getting PTY: %v", err)
+			return 0
+		}
+
+		cmd.Stdin = s
+
+		// Launch go routine to copy output from the command to the PTM
+		for _, r := range cmd.Env {
+			if strings.HasPrefix(r, "tty") {
+				tty := strings.Split(r, "=")[1]
+
+				debug("Opening TTY %v", tty)
+
+				// Open the TTY
+				t, err := os.OpenFile(tty, os.O_RDWR, 0)
+				if err != nil {
+					log.Printf("Error opening TTY: %v", err)
+					continue
+				}
+
+				// Set to Raw mode
+				if err := Raw(t); err != nil {
+					// Let's still continue if we can't set the TTY to raw mode
+					log.Printf("Error setting TTY to raw mode: %v", err)
+				}
+
+				go func() {
+					for {
+						_, err := io.Copy(m, t)
+						if err != nil {
+							log.Printf("Error copying output from command to PTM: %v", err)
+						}
```

Comment:
you might consider passing in an io.Writer for the package to log errors to, I can see where this kind of logging might be useful, but just directly printing like this can really mess up output.